daemon/container.go

```diff
@@ -576,6 +581,64 @@ func validateID(id string) error {
 	return nil
 }
 
+func (container *Container) Checkpoint(opts *libcontainer.CriuOpts) error {
+	if err := container.daemon.Checkpoint(container, opts); err != nil {
+		return err
+	}
+
+	if opts.LeaveRunning == false {
```

Comment:
I actually think we should remove this. There's a related section in cleanup that's not releasing when checkpointed. I think that maybe can get removed.